src/librbd/mirror/snapshot/CreatePrimaryRequest.cc

```diff
@@ -208,16 +209,18 @@ void CreatePrimaryRequest<I>::unlink_peer() {
       count++;
       if (count == 3) {
         unlink_snap_id = snap_it.first;
+        if (info->mirror_peer_uuids.size() == 0) {
```

Comment:
@MrFreezeex Isn't the problem actually that we don't check that `peer` exists in `info->mirror_peer_uuids`? And as a result always call `UnlinkPeerRequest` for the first peer in `m_mirror_peer_uuids`, even if it is not here, going into the loop?

If my understanding is correct, then it looks like the fix could be just to check that `peer` is in `info->mirror_peer_uuids` list, and if it is not continue with the next snapshot?

Reply:
Yes your assumption is correct, we could add a condition like that to the code and add a `continue` but this may not preserve the order of deletion (if there is more than 1 snapshot to delete somehow). So maybe a `break` would be more appropriate to re-do the loop and then select the same snapshot and check again if you are on the right loop to send an UnlinkRequest against it.

This whole thing makes me think that there is not a real need for a loop, but I do one of the approach described with a `continue` or a `break` if you prefer.

Reply:
I don't see any issue with the order of deletion. The order of unlinking a peer is probably important, and it is ok here.

So I would try to make the fix as simple as possible. But if you still prefer your way (no need for loop), this seems ok to me, if I understand what you mean, so I need to look at the code. Just update the PR as you like it and I will look.